src/version.c

```diff
@@ -34,36 +34,43 @@
 
 #define MAX_VERSION_STR_SIZE 11
 
-int get_server_format(int server_version)
+int get_int_from_url(const char *url)
 {
-	if (server_version < 0) {
-		return -1;
-	}
-
-	int ret, err = -1;
-	char *url;
-
-	char format_string[MAX_VERSION_STR_SIZE];
-	struct curl_file_data format = {
+	int err, value;
+	char tmp_string[MAX_VERSION_STR_SIZE];
+	struct curl_file_data tmp_data = {
 		MAX_VERSION_STR_SIZE, 0,
-		format_string
+		tmp_string
 	};
 
-	string_or_die(&url, "%s/%d/format", globals.version_url, server_version);
-	ret = swupd_curl_get_file_memory(url, &format);
-	free_string(&url);
-
-	if (ret) {
+	value = swupd_curl_get_file_memory(url, &tmp_data);
+	if (value) {
 		return -1;
 	}
 
-	format.data[format.len] = '\0';
-	err = strtoi_err(format.data, &ret);
+	tmp_data.data[tmp_data.len] = '\0';
+	err = strtoi_err(tmp_data.data, &value);
 	if (err != 0) {
 		return -1;
 	}
 
-	return ret;
+	return value;
+}
+
+int get_server_format(int server_version)
```

Comment:
same thing here with the name, although in this case a name that starts with "version" doesn't make sense: `version_get_serveer_format`. This brings the question, does this function belongs in this file? Maybe it should go in the helpers.c file where we have more general functions. It could keep the current name in that file.

Reply:
In general functions exported in swupd.h are undocumented, not following a good name system and spread in .c that not necessary make sense. We need to try to organize them, as we improve the quality, but I usually don't concern that much with them, unless it's something significant that is being changed.